src/callbacks_stage/subcell_limiter_idp_correction_2d.jl

```diff
@@ -11,27 +11,48 @@ function perform_idp_correction!(u, dt,
                                                      mesh, i, j, element)
 
-            # Note: antidiffusive_flux1[v, i, xi, element] = antidiffusive_flux2[v, xi, i, element] = 0 for all i in 1:nnodes and xi in {1, nnodes+1}
-            alpha_flux1 = (1 - alpha1[i, j, element]) *
-                          get_node_vars(antidiffusive_flux1_R, equations, dg,
-                                        i, j, element)
-            alpha_flux1_ip1 = (1 - alpha1[i + 1, j, element]) *
-                              get_node_vars(antidiffusive_flux1_L, equations, dg,
-                                            i + 1, j, element)
-            alpha_flux2 = (1 - alpha2[i, j, element]) *
-                          get_node_vars(antidiffusive_flux2_R, equations, dg,
-                                        i, j, element)
-            alpha_flux2_jp1 = (1 - alpha2[i, j + 1, element]) *
-                              get_node_vars(antidiffusive_flux2_L, equations, dg,
-                                            i, j + 1, element)
+            # Note: For LGL nodes, the high-order and low-order fluxes at element interfaces are equal.
+            # Therefore, we only need to consider the inner updates and setting the outer fluxes to zero.
+            if i > 1
+                alpha1 = max(alpha[i - 1, j, element], alpha[i, j, element])
+                alpha_flux1 = (1 - alpha1) *
+                              get_node_vars(antidiffusive_flux1_R, equations, dg,
+                                            i, j, element)
+            else
+                alpha_flux1 = zero(SVector{nvariables(equations), eltype(u)})
+            end
+            if i < nnodes(dg)
+                alpha1_ip1 = max(alpha[i, j, element], alpha[i + 1, j, element])
+                alpha_flux1_ip1 = (1 - alpha1_ip1) *
+                                  get_node_vars(antidiffusive_flux1_L, equations, dg,
+                                                i + 1, j, element)
+            else
+                alpha_flux1_ip1 = zero(SVector{nvariables(equations), eltype(u)})
+            end
+            if j > 1
+                alpha2 = max(alpha[i, j - 1, element], alpha[i, j, element])
+                alpha_flux2 = (1 - alpha2) *
+                              get_node_vars(antidiffusive_flux2_R, equations, dg,
+                                            i, j, element)
+            else
+                alpha_flux2 = zero(SVector{nvariables(equations), eltype(u)})
+            end
+            if j < nnodes(dg)
+                alpha2_jp1 = max(alpha[i, j, element], alpha[i, j + 1, element])
+                alpha_flux2_jp1 = (1 - alpha2_jp1) *
+                                  get_node_vars(antidiffusive_flux2_L, equations, dg,
+                                                i, j + 1, element)
+            else
+                alpha_flux2_jp1 = zero(SVector{nvariables(equations), eltype(u)})
+            end
```

Comment:
If anyone has a better idea for this, I'd be open to it. I just wanted to keep the flux-differencing update below. So, I needed those fluxes even in the zero case.

Reply:
The only negative point would be that we have the structure with `nnodes(dg) + 1` for `fhat` and `f_star`. So, the indices would be different then:/

Reply:
After the last argument, I would actually prefer the suggested version from above https://github.com/trixi-framework/Trixi.jl/pull/2588#discussion_r2381267583.
So, get rid of the `alpha1/2`, but keep the structure of the antidiffusive fluxes.
The only open thing is whether it is better to use the zeros from the vector or set new zeros as it is done now.

Reply:
I adapted the comment above and are now happy with the current state.